query_utils.py

```diff
@@ -50,33 +57,34 @@ async def query_openai(
     for i in range(retry + 1):
         wait_time = (1 << min(i, OPENAI_EXP_CAP)) + random() / 10
         try:
+            logger.info("Attempting API call...")
```

Comment:
I don't use the logging library much. What's the *default* logging level? If it's INFO, then this might be too noisy. But if it's >INFO, this should be fine.

Reply:
The default logging level is WARNING/ERROR. It's set to INFO because log messages won't be displayed by default.